pandas/io/excel/_base.py

```diff
@@ -165,31 +165,12 @@
     Supported engines: "xlrd", "openpyxl", "odf", "pyxlsb", "calamine".
     Engine compatibility :
 
-    - ``xlr`` supports old-style Excel files (.xls).
     - ``openpyxl`` supports newer Excel file formats.
-    - ``odf`` supports OpenDocument file formats (.odf, .ods, .odt).
-    - ``pyxlsb`` supports Binary Excel files.
     - ``calamine`` supports Excel (.xls, .xlsx, .xlsm, .xlsb)
       and OpenDocument (.ods) file formats.
-
-    .. versionchanged:: 1.2.0
-        The engine `xlrd <https://xlrd.readthedocs.io/en/latest/>`_
-        now only supports old-style ``.xls`` files.
-        When ``engine=None``, the following logic will be
-        used to determine the engine:
```

Comment:
I do think it's good to have this logic documented, can you just move it out of the versionchanged instead?

Reply:
IIRC xlrd used to not only support xlsx files but at one point was even the default so we had to go through some lengths to document that transition as clearly as possible. We are a few years removed from that and since then all default read libraries have specialized in a given extension(s), so I _think_ we can do without going into this detail in the docstring

Reply:
> and since then all default read libraries have specialized in a given extension(s)

What are the default read libraries and where are they documented? E.g. both openpyxl and calamine can read xlsx, and both pyxlsb and calamine can read xlsb.

Reply:
Calamine is never the default - you'd have to explicitly use that as an engine. Otherwise this is documented in the Excel section of the IO manual

https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#excel-files

Reply:
Granted that section could be rewritten to be a little clearer, but I think that is out of scope for what @phofl is doing here

Reply:
> Calamine is never the default - you'd have to explicitly use that as an engine.

This is true today, but there is an issue to make it the default for xlsb files. In any case, I don't believe it's documented that Calamine is never the default.

Reply:
Updated and added back in without the version changed